modules/RTC/TraceablePeerConnection.js

```diff
@@ -910,68 +1639,149 @@ TraceablePeerConnection.prototype.createAnswer
                 //  details)
-                const remoteDescription = new SDP(this.remoteDescription.sdp);
-                const localDescription = new SDP(answer.sdp);
+                if (!isOffer) {
+                    const remoteDescription
+                        = new SDP(this.remoteDescription.sdp);
+                    const localDescription = new SDP(resultSdp.sdp);
+
+                    _fixAnswerRFC4145Setup(remoteDescription, localDescription);
+                    resultSdp.sdp = localDescription.raw;
+                }
+
+                const ssrcMap = extractSSRCMap(resultSdp);
 
-                _fixAnswerRFC4145Setup(remoteDescription, localDescription);
-                answer.sdp = localDescription.raw;
+                logger.info('Got SSRC MAP: ', ssrcMap);
 
-                this.eventEmitter.emit(RTCEvents.SENDRECV_STREAMS_CHANGED,
-                    extractSSRCMap(answer));
+                // Set up the ssrcHandler for the new track before we add it at
+                // the lower levels
+                this._applyLocalSSRCMap(ssrcMap);
 
-                successCallback(answer);
+                successCallback(resultSdp);
             } catch (e) {
-                this.trace('createAnswerOnError', e);
-                this.trace('createAnswerOnError', dumpSDP(answer));
-                logger.error('createAnswerOnError', e, dumpSDP(answer));
+                this.trace(`create${logName}OnError`, e);
+                this.trace(`create${logName}OnError`, dumpSDP(resultSdp));
+                logger.error(
+                    `create${logName}OnError`, e, dumpSDP(resultSdp));
                 failureCallback(e);
             }
         },
         err => {
-            this.trace('createAnswerOnFailure', err);
-            this.eventEmitter.emit(RTCEvents.CREATE_ANSWER_FAILED, err,
-                this.peerconnection);
+            this.trace(`create${logName}OnFailure`, err);
+            const eventType
+                = isOffer
+                    ? RTCEvents.CREATE_OFFER_FAILED
+                    : RTCEvents.CREATE_ANSWER_FAILED;
+
+            this.eventEmitter.emit(eventType, err, this.peerconnection);
             failureCallback(err);
         },
         constraints);
-    };
+};
 
-TraceablePeerConnection.prototype.addIceCandidate
+/* eslint-enable max-params */
+
+/**
+ *
+ * @param {TrackSSRCInfo} ssrcObj
+ * @return {number|null}
+ */
+function extractPrimarySSRC(ssrcObj) {
+    if (ssrcObj && ssrcObj.groups && ssrcObj.groups.length) {
+        return ssrcObj.groups[0].ssrcs[0];
+    } else if (ssrcObj && ssrcObj.ssrcs && ssrcObj.ssrcs.length) {
+        return ssrcObj.ssrcs[0];
+    }
+
+    return null;
+}
+
+/**
+ * Applies SSRC map extracted from the latest local description to local tracks.
+ * @param {Map<string,TrackSSRCInfo>} ssrcMap
+ * @private
+ */
+TraceablePeerConnection.prototype._applyLocalSSRCMap = function(ssrcMap) {
```

Comment:
Yes, but SDPConsistency only handles video and 1 track. When I added support for storing local tracks in TPC there was a need to store that for any track. Maybe some cleanup around that will be required.